src/lib/work_partitioner/work_partitioner.ml

```diff
@@ -348,7 +347,21 @@ let submit_single ~partitioner
       | `Removed ->
           Removed
       | `SpecUnmatched ->
+          Option.iter removed_job ~f:(fun job ->
+              Sent_single_job_pool.bring_back ~id:job_id ~job
```

Comment:
This is so if snark worker fails on providing a proof, put that back to single job pool to be scheduled again.